### PR TITLE
[OCNative] remove get_api_resources

### PR DESCRIPTION
**Pull request by maorfr**
related to https://issues.redhat.com/browse/APPSRE-6438

OCNative should only handle GET requests. This removed function is long running and because OCDeprecated (should be OCCli) is doing everything else, there is little to no sense to get api-resources this way (especially when it takes a very long time).

### CODE REVIEW

**Review by chassing — APPROVED** [review]
RIP `get_api_resources` 💀

**Comment by BumbleFeng** [comment]
depends on https://github.com/app-sre/qontract-reconcile/pull/3367

File: reconcile/test/test_utils_oc_native.py
Comment (BumbleFeng) [inline reconcile/test/test_utils_oc_native.py:1]: please also remove https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/test/fixtures/oc_native/api.yml

Comment (maorfr) [reply reconcile/test/test_utils_oc_native.py:1]: https://github.com/app-sre/qontract-reconcile/pull/3351/commits/b898933f3d90b89c8294692e00ca5b7fe15c1c01

File: reconcile/utils/oc.py
Comment (BumbleFeng) [inline reconcile/utils/oc.py:1]: need to remove unused import https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/oc.py#L33

Comment (maorfr) [reply reconcile/utils/oc.py:1]: https://github.com/app-sre/qontract-reconcile/pull/3351/commits/dc1a6007a44d53cdb8228a2f002e2f04a649959d

**Comment by hemslo** [comment]
This one failed because we still need to get `(kind, group_version)` through [_parse_kind](https://github.com/app-sre/qontract-reconcile/blob/a538da70ccd63ccdb98fcc075f3858d1c0b4c06b/reconcile/utils/oc.py#L1128), which is rely on `api_resources`, it's being used in multiple methods in `OCNative`, like [get_items](https://github.com/app-sre/qontract-reconcile/blob/a538da70ccd63ccdb98fcc075f3858d1c0b4c06b/reconcile/utils/oc.py#L1341).

`is_kind_supported` is a public method, if we remove it, it will fallback to the parent method [is_kind_supported](https://github.com/app-sre/qontract-reconcile/blob/a538da70ccd63ccdb98fcc075f3858d1c0b4c06b/reconcile/utils/oc.py#L1162), which is also relying on `api_resources`.

**Pull request merged**